spec/prog/vnet/rekey_nic_spec.rb

```diff
@@ -0,0 +1,206 @@
+# frozen_string_literal: true
+
+RSpec.describe Prog::Vnet::RekeyNic do
+  subject(:nx) {
+    described_class.new(st)
+  }
+
+  let(:st) { Strand.new }
+  let(:ps) {
+    PrivateSubnet.create_with_id(name: "ps", location: "hetzner-hel1", net6: "fd10:9b0b:6b4b:8fbb::/64",
+      net4: "1.1.1.0/26", state: "waiting")
+  }
+  let(:tunnel) {
+    n_src = Nic.create_with_id(private_subnet_id: ps.id,
+      private_ipv6: "fd10:9b0b:6b4b:8fbb:abc::",
+      private_ipv4: "10.0.0.1",
+      mac: "00:00:00:00:00:00",
+      encryption_key: "0x736f6d655f656e6372797074696f6e5f6b6579",
+      name: "default-nic")
+    n_dst = Nic.create_with_id(private_subnet_id: ps.id,
+      private_ipv6: "fd10:9b0b:6b4b:8fbb:def::",
+      private_ipv4: "10.0.0.2",
+      mac: "00:00:00:00:00:00",
+      encryption_key: "0x736f6d655f656e6372797074696f6e5f6b6579",
+      name: "default-nic")
+    IpsecTunnel.create_with_id(src_nic_id: n_src.id, dst_nic_id: n_dst.id).tap { _1.id = "0a9a166c-e7e7-4447-ab29-7ea442b5bb0e" }
+  }
+
+  before do
+    nx.instance_variable_set(:@nic, tunnel.src_nic)
+  end
+
+  describe ".sshable_cmd" do
+    let(:sshable) { instance_double(Sshable) }
+    let(:vm) {
+      vmh = instance_double(VmHost, sshable: sshable)
+      instance_double(Vm, vm_host: vmh)
+    }
+
+    it "returns vm_host sshable of source nic" do
+      expect(nx.nic).to receive(:vm).and_return(vm)
+      expect(sshable).to receive(:cmd).with("echo hello")
+      nx.sshable_cmd("echo hello")
+    end
+  end
+
+  describe "#start" do
+    let(:vm) {
+      instance_double(Vm, name: "hellovm", id: "0a9a166c-e7e7-4447-ab29-7ea442b5bb0e",
+        ephemeral_net6: NetAddr.parse_net("2a01:4f8:10a:128b:4919::/80"))
+    }
+
+    before do
+      expect(nx).to receive(:frame).and_return(
+        {
+          "payload" => {tunnel.src_nic.id =>
```

Comment:
It's interesting that you wrap stuff in a payload, normally when passing args it's flat, any special justification here for the nesting?

Reply:
oooh it's so you can feed a smaller, well defined part of the frame into args, man, you really swiss-watched this thing in its concision.

Reply:
now it's part of the entity with the new pr https://github.com/ubicloud/ubicloud/pull/441